src/Aspire.Dashboard/Components/Pages/IComponentWithTelemetry.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Telemetry;
+
+namespace Aspire.Dashboard.Components.Pages;
+
+/// <summary>
+/// Interface for components that would like to opt-in to telemetry events, capturing
+/// 1) initialization
+/// 2) component properties on each render
+/// 3) component disposal
+/// </summary>
+public interface IComponentWithTelemetry
+{
+    public DashboardTelemetryService TelemetryService { get; }
+    public ComponentTelemetryContext TelemetryContext { get; }
+
+    public void UpdateTelemetryProperties()
+    {
+    }
+}
+
+public static class IComponentWithTelemetryExtensions
+{
+    public static async Task InitializeComponentTelemetryAsync(this IComponentWithTelemetry component)
+    {
+        await component.TelemetryService.InitializeAsync();
+
+        component.TelemetryContext.InitializeCorrelation = component.TelemetryService.PostUserTask(
+            TelemetryEventKeys.ComponentInitialize,
+            TelemetryResult.Success,
+            properties: new Dictionary<string, AspireTelemetryProperty>
+            {
+                // Component properties
+                { TelemetryPropertyKeys.DashboardComponentId, new AspireTelemetryProperty(component.TelemetryContext.ComponentType) }
+            });
+    }
+
+    public static void DisposeComponentTelemetry(this IComponentWithTelemetry component)
```

Comment:
What happens here if there is an exception in the component's initialize method, and `InitializeComponentTelemetryAsync` is never called?